BUILD

```diff
@@ -39,6 +39,23 @@ java_library(
     ],
 )
 
+load("//tools:jarjar.bzl", "jarjar_library")
+
+genrule(
+    name = "rules_file",
+    outs = ["rules_file.txt"],
+    cmd = "echo \"rule com.google.auto.common.** dagger.shaded.auto.common.@1\" > $@",
+)
+
+jarjar_library(
+    name = "shaded_compiler",
+    rules_file = ":rules_file.txt",
+    deps = [
+        "//compiler",
+        "@com_google_auto_auto_common//jar",
```

Comment:
This should probably shade (and relocate) JavaPoet (JavaPoet is shaded in the Maven build AFAICT, but not relocated)

The Maven build also shades google-java-format, errorprone annotations, JSR305 annotations and errorprone-javac (via google-java-format), without relocating them.